ml-prims/src/linalg/coalesced_reduction.h

```diff
@@ -58,10 +68,19 @@ __global__ void coalescedReductionKernel(Type *dots, const Type *data, int D, in
 /**
  * @brief Compute reduction of the input matrix along the leading dimension
  *
- * @tparam Type the data type
+ * @tparam InType the data type of the input
+ * @tparam OutType the data type of the output (as well as the data type for
+ *  which reduction is performed)
+ * @tparam IdxType data type of the indices of the array
  * @tparam MainLambda Unary lambda applied while acculumation (eg: L1 or L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*MainLambda)(InType, IdxType);</pre>
  * @tparam ReduceLambda Binary lambda applied for reduction (eg: addition(+) for L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*ReduceLambda)(OutType);</pre>
  * @tparam FinalLambda the final lambda applied before STG (eg: Sqrt for L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*ReduceLambda)(OutType);</pre>
```

Comment:
Should be `OutType (*FinalLambda)(OutType)`

Reply:
Fixed.